cmd/snap-exec/main.go

```diff
@@ -146,9 +151,16 @@ func snapExecApp(snapApp, revision, command string, args []string) error {
 
 	// run the command
 	fullCmd := filepath.Join(app.Snap.MountDir(), cmd)
-	if command == "shell" {
+	switch command {
+	case "shell":
 		fullCmd = "/bin/bash"
 		cmdArgs = nil
+	case "complete":
+		fullCmd = "/bin/bash"
+		cmdArgs = []string{
+			dirs.CompletionHelper,
+			filepath.Join(app.Snap.MountDir(), app.Completer),
```

Comment:
snap-exec always has `app.Snap.MountDir()` to `/snap/...`, which is correct because it's always "inside".